pkg/controllers/dnshealthcheckprobe/dnshealthcheckprobe_controller.go

```diff
@@ -92,8 +92,15 @@ func (r *DNSHealthCheckProbeReconciler) Reconcile(ctx context.Context, req ctrl.
 	additionalHeaders, err := getAdditionalHeaders(ctx, r.Client, probeObj)
 
 	if err != nil {
-		//update probe status, ignore update errors
-		_ = r.Client.Status().Update(ctx, probeObj)
+		f := false
+		logger.Error(err, "error getting additional headers for probe")
```

Comment:
Given the status is available to the user,
and the cause of the error is very likely something the user can/should fix themselves,
I wonder if 'error' level logging is overkill here.
If I was an ops or sre person running this service, I would expect errors to be something I have to react to.
In the case of a missing secret, I can't fix that. The status information on the resource is how the user gets notified.